File: .claude/hooks/notification.py
import json
import os
import sys
import urllib.request
import urllib.parse
from datetime import datetime

def send_telegram_message(bot_token, chat_id, message):
    """Send a Telegram message"""
    url = f"https://api.telegram.org/bot{bot_token}/sendMessage"
    data = {
        'chat_id': chat_id,
        'text': message,
        'parse_mode': 'Markdown'
    }
    
    try:
        req = urllib.request.Request(
            url,
            data=urllib.parse.urlencode(data).encode('utf-8'),
            headers={'Content-Type': 'application/x-www-form-urlencoded'}
        )
        with urllib.request.urlopen(req, timeout=5) as response:
            return response.status == 200
    except Exception:
        return False

def format_message(event_data, project_name):
    """Format event data into a Telegram message"""
    event = event_data.get('hook_event_name', 'Unknown')
    stop_hook_active = event_data.get('stop_hook_active', False)
    timestamp = datetime.now().strftime('%Y-%m-%d %H:%M:%S')
    
    # Base message format
    message = f"🤖 *Project: `{project_name}`*\n"
    message += f"⏰ {timestamp}\n"
    message += f"✅ Event: `{event}`\n"
    message += f"📌 Stop Hook Active: `{stop_hook_active}`"
    
    return message

def main():
    try:
        # Telegram configuration
        bot_token = os.getenv('CC_HOOK_TELEGRAM_BOT_TOKEN')
        chat_id = os.getenv('CC_HOOK_TELEGRAM_CHAT_ID')
        
        # Read input data
        input_data = json.loads(sys.stdin.read().strip())

        current_dir = os.getcwd()
        project_name = os.path.basename(current_dir)
        
        if input_data:
            # Format message
            message = format_message(input_data, project_name)
            
            # Send to Telegram
            send_telegram_message(bot_token, chat_id, message)
    
    except json.JSONDecodeError:
        sys.exit(0)
    except Exception:
        sys.exit(0)
    finally:
        sys.exit(0)

if __name__ == "__main__":
    main()

File: .claude/hooks/notification.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"net/url"
	"os"
	"path/filepath"
	"strings"
	"time"
)

var (
	botToken = os.Getenv("CC_HOOK_TELEGRAM_BOT_TOKEN")
	chatID   = os.Getenv("CC_HOOK_TELEGRAM_CHAT_ID")
)


func sendTelegramMessage(message string) error {
	apiURL := fmt.Sprintf("https://api.telegram.org/bot%s/sendMessage", botToken)
	
	data := url.Values{}
	data.Set("chat_id", chatID)
	data.Set("text", message)
	data.Set("parse_mode", "HTML")
	
	client := &http.Client{Timeout: 5 * time.Second}
	resp, err := client.Post(apiURL, "application/x-www-form-urlencoded", strings.NewReader(data.Encode()))
	if err != nil {
		return err
	}
	defer resp.Body.Close()
	
	if resp.StatusCode != http.StatusOK {
		return fmt.Errorf("telegram API returned status %d", resp.StatusCode)
	}
	
	return nil
}


func formatMessage(rawData map[string]interface{}) string {
	// 프로젝트 이름 추출
	projectName := "Unknown"
	if cwd, ok := rawData["cwd"].(string); ok && cwd != "" && cwd != "Unknown" {
		projectName = filepath.Base(cwd)
	}
	
	// 이벤트 이름 추출
	eventName := "Unknown"
	if event, ok := rawData["hook_event_name"].(string); ok {
		eventName = event
	}

	eventMessage := "Unknown"
	if msg, ok := rawData["message"].(string); ok {
		eventMessage = msg
	}

	needPermission := strings.HasPrefix(eventMessage, "Claude needs your permission")

	timestamp := time.Now().Format("2006-01-02 15:04:05")
	
	// 메시지 작성 (HTML 형식 사용)
	message := fmt.Sprintf("🤖 <b>Project: %s</b>\n", projectName)
	message += fmt.Sprintf("⏰ %s\n", timestamp)
	message += fmt.Sprintf("✅ Event: <code>%s</code>\n", eventName)
	if needPermission {
		message += fmt.Sprintf("📌 <b>Need Permission:</b> <code>%v</code>\n", needPermission)
	}
	
	return message
}

func sendErrorNotification(errorMsg string) {
	// 현재 디렉토리에서 프로젝트 이름 가져오기
	cwd, _ := os.Getwd()
	projectName := filepath.Base(cwd)
	
	timestamp := time.Now().Format("2006-01-02 15:04:05")
	
	// HTML 형식 사용
	errorMessage := "🚨 <b>Hook Error Alert</b>\n"
	errorMessage += fmt.Sprintf("📁 <b>Project:</b> <code>%s</code>\n", projectName)
	errorMessage += fmt.Sprintf("⏰ %s\n", timestamp)
	errorMessage += fmt.Sprintf("❌ <b>Error Details:</b>\n<pre>%s</pre>", errorMsg)
	
	_ = sendTelegramMessage(errorMessage)
}

func main() {
	// stdin에서 JSON 데이터 읽기
	inputData, err := io.ReadAll(os.Stdin)
	if err != nil {
		sendErrorNotification(fmt.Sprintf("Failed to read stdin: %v", err))
		os.Exit(1)
	}
	
	// 빈 입력 처리
	if len(bytes.TrimSpace(inputData)) == 0 {
		os.Exit(0)
	}
	
	// JSON 파싱 - 전체 데이터를 먼저 map으로 파싱
	var rawData map[string]interface{}
	if err := json.Unmarshal(inputData, &rawData); err != nil {
		sendErrorNotification(fmt.Sprintf("JSON Decode Error: %v\nInput: %s", err, string(inputData)))
		os.Exit(1)
	}
	
	// 메시지 포맷팅 및 전송
	message := formatMessage(rawData)
	if err := sendTelegramMessage(message); err != nil {
		sendErrorNotification(fmt.Sprintf("Failed to send telegram message: %v", err))
		os.Exit(1)
	}
	
	os.Exit(0)
}